### PR TITLE
Add ExternalSecret CR for snyk_token

### PR DESCRIPTION
**Pull request by navidshaikh**
- This CR brings snyk_token defined in google secret manager
   into test infra for running periodic snyk CI job.
 - this secret is referred in https://github.com/kubernetes/test-infra/pull/22293

xref:  https://github.com/kubernetes/kubernetes/issues/101528
 
/cc: @cjwagner @spiffxp

### CODE REVIEW

File: config/prow/cluster/kubernetes_external_secrets.yaml [hunk 1]
```diff
@@ -39,3 +39,16 @@ spec:
   - key: gke_k8s-prow_us-central1-f_prow__prow-monitoring__grafana
     name: password
     version: latest
+---
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: snyk_token
+  namespace: prow-test
```
Comment (navidshaikh) [inline config/prow/cluster/kubernetes_external_secrets.yaml:47]: is this correct namespace where periodic prow jobs are run ?  (https://github.com/kubernetes/test-infra/pull/22293)

Comment (navidshaikh) [reply config/prow/cluster/kubernetes_external_secrets.yaml:47]: A: `test-pods` (also populated in env var `PROWJOB_POD_NAMESPACE`)
see https://github.com/kubernetes/k8s.io/pull/2078

**Review by cjwagner — CHANGES_REQUESTED** [review]
If this secret is for a job in the default build cluster then it needs to be defined in [`config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml`](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml). The file you updated in this PR would incorrectly add the secret to the Prow service cluster. Instructions for setting up secrets for the default build cluster can be found here: https://github.com/kubernetes/test-infra/blob/master/prow/prow_secrets.md#usage-prow-clients

**Comment by spiffxp** [comment]
FWIW I am working to get the k8s-infra-prow-build-trusted cluster setup with kubernetes-external-secrets. Perhaps we can run there?

File: config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml [hunk 2]
```diff
@@ -13,3 +13,16 @@ spec:
...
     version: latest
+---
+apiVersion: kubernetes-client.io/v1
+kind: ExternalSecret
+metadata:
+  name: snyk_token
+  namespace: test-pods
+spec:
+  backendType: gcpSecretsManager
+  projectId: kubernetes-upstream # gsm project id
```
Comment (navidshaikh) [inline config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml:24]: verify if this projectId is correct

/hold

Comment (navidshaikh) [reply config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml:24]: cc: @spiffxp @cjwagner is this correct project we'd want to have secrets into?

Comment (spiffxp) [reply config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml:24]:
```suggestion
  projectId: k8s-infra-prow-build-trusted # gsm project id
```

Comment (navidshaikh) [reply config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml]: done
/unhold

Comment (navidshaikh) [reply config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml]: @spiffxp : can this CR reside in this file ? or should I move to https://github.com/kubernetes/k8s.io/tree/main/infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources ?

File: config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml [hunk 3]
```diff
@@ -13,3 +13,16 @@ spec:
...
+metadata:
+  name: snyk-token
+  namespace: test-pods
+spec:
+  backendType: gcpSecretsManager
+  projectId: k8s-infra-prow-build-trusted # gsm project id
+  data:
+  - key: snyk-token  # name of the GCP secret
+    name: SNYK_TOKEN  # key name in the k8s secret
+    version: latest
```
Comment (spiffxp) [inline config/prow/cluster/build_kubernetes-external-secrets_customresource.yaml:28]: This is going to need to be redone as a PR to https://github.com/kubernetes/k8s.io/blob/main/infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/test-pods-externalsecrets.yaml

Let me check on the job definition PR

**Comment by spiffxp** [comment]
/close
I'm going to redo this PR over in k8s.io

**Pull request closed without merging**